File: src/utils/prompt.rs
use anyhow::Result;
use console::style;
use dialoguer::{Confirm, Input, Select};

/// Print a step header
pub fn step(num: u32, total: u32, title: &str) {
    println!(
        "\n{} {}",
        style(format!("[{}/{}]", num, total)).cyan().bold(),
        style(title).bold()
    );
}

/// Print a success message
pub fn success(msg: &str) {
    println!("  {} {}", style("✓").green(), msg);
}

/// Print an info message
pub fn info(msg: &str) {
    println!("  {} {}", style("→").blue(), msg);
}

/// Print a warning message
pub fn warn(msg: &str) {
    println!("  {} {}", style("⚠").yellow(), msg);
}

/// Ask for confirmation
pub fn confirm(msg: &str, default: bool) -> Result<bool> {
    Ok(Confirm::new()
        .with_prompt(msg)
        .default(default)
        .interact()?)
}

/// Ask for confirmation, or return true if --yes was passed
pub fn confirm_or_yes(msg: &str, default: bool, yes: bool) -> Result<bool> {
    if yes {
        Ok(true)
    } else {
        confirm(msg, default)
    }
}

/// Ask for text input with a default value
pub fn input(prompt: &str, default: &str) -> Result<String> {
    Ok(Input::new()
        .with_prompt(prompt)
        .default(default.to_string())
        .interact_text()?)
}

/// Select from a list of options
pub fn select(prompt: &str, options: &[&str], default: usize) -> Result<usize> {
    Ok(Select::new()
        .with_prompt(prompt)
        .items(options)
        .default(default)
        .interact()?)
}

/// Print a section header
pub fn section(title: &str) {
    println!("\n{}", style(title).bold().underlined());
}

/// Print a key-value pair
pub fn kv(key: &str, value: &str) {
    println!("  {}: {}", style(key).dim(), value);
}

File: src/wslarc/utils/prompt.py
"""Small terminal prompt and output helpers."""

from __future__ import annotations


def confirm(message: str, default: bool = False) -> bool:
    suffix = " [Y/n] " if default else " [y/N] "
    answer = input(message + suffix).strip().lower()
    if not answer:
        return default
    return answer in {"y", "yes"}


def confirm_or_yes(message: str, default: bool, yes: bool) -> bool:
    return True if yes else confirm(message, default)


def input_default(message: str, default: str = "") -> str:
    answer = input(f"{message} [{default}] ").strip()
    return answer or default


def info(message: str) -> None:
    print(f"  → {message}")


def success(message: str) -> None:
    print(f"  ✓ {message}")


def warn(message: str) -> None:
    print(f"  ⚠ {message}")


def step(number: int, total: int, message: str) -> None:
    print(f"\n[{number}/{total}] {message}")


def section(title: str) -> None:
    print(f"\n{title}\n{'=' * len(title)}")


def kv(key: str, value: str) -> None:
    print(f"  {key}: {value}")
